0004-post-tags.md

```diff
@@ -0,0 +1,324 @@
+```json
+{
+  id: number,
+  url: string,
+  name: string
+}
+```
+
+Entirely unmoderated tags are not an option for lemmy as the moderation workload would be too much. Additionally users being able to type out tags themselves introduces splintering in the tag contents due to typos. A better solution is a curated list of tags users can attach to their posts. The list of tags can be maintained by both admins and moderators allowing for each community to tailor tags to their specific needs.
+Content for tags would be located in the respective root: `https://example.org/c/example/t/tag`
+
+The tag URL could then be utilized as a unique identifier for the tag, however doing so would restrict the ability to move the underlying tag url in the future. Instead an additional `id` field should be used. Using the object ID optional tag federation can also be achieved, allowing for communities across multiple instances to share content via tags (example: News tag shared across instances). This would also solve the issue of splintered communities across instances while not forcing it on the communities in question. For now tags will only be applicable to posts, however the general design allows for them to be attached to any kind of object later on, be that instance, community or user. The limited initial scope allows for easier modifications should any rough edges or missing features be discovered.
+
+To fill all needed use cases 3 tag flavors will be needed:
+
+- NSFW
+- Content Warning
+- Generic Tag
+
+Theoretically NSFW could be implemented using a preset "Content Warning" tag but seperating out this tag allows instances to better filter it out for moderation purposes (for example if no admin/moderator is willing to moderate NSFW content).
+Both "NSFW" and "Content Warning" tags should blur the post body by default. Additionally the `sensitive` post flag to `true` should either of these flavors be present in the post tags to ensure correct handling on other fediverse platforms.
+
+Example Tag Objects:
+
+**Generic Tag:**
+```json
+{
+    "id": 1,
+    "url": "https://example.org/c/community/t/tag",
+    "name": "Generic Tag"
+}
+```
+
+**NSFW Tag:**
+```json
+{
+    "flavor": "nsfw",
+    "id": 1,
+    "url": "https://example.org/c/adult_community/t/nsfw",
+    "name": "NSFW"
+}
+```
+
+**Spoiler Tag:**
+```json
+{
+    "flavor": "cw",
+    "id": 1,
+    "url": "https://example.org/c/story_community/t/spoiler",
+    "name": "Spoiler"
+}
+```
+
+**Remote Instance Spoiler Tag:**
+```json
+{
+    "flavor": "cw",
+    "id": 1,
+    "url": "https://example.org/c/example_community/tag@example_remote.org",
+    "name": "Spoiler"
+}
+```
+
+Below is an example of a News post in the news community. The post has a communtiy tag for the newspaper company that published it, as well as a content warning for Blood/Gore. In order to share this news post across instances it also is tagged with the news tag of a remote instance.
+
+Example Post json:
+
+``` json
+{
```

Comment:
just add context to make the object actually globally well-defined, and add the community for clarity:

```suggestion
{
  "@context": [
    "https://www.w3.org/ns/activitystreams",
    "https://join-lemmy.org/context.json"
  ],
  "to": [
    "https://example.org/c/news",
    "https://www.w3.org/ns/activitystreams#Public"
  ],
  "audience": "https://example.org/c/news",
```